PerformanceTests/MotionMark/resources/debug-runner/motionmark.js

```diff
@@ -679,5 +682,44 @@ Utilities.extendObject(window.benchmarkController, {
         sectionsManager.setSectionHeader("test-graph", testName);
         sectionsManager.showSection("test-graph", true);
         this.updateGraphData(testResult, testData, benchmarkRunnerClient.results.options);
+    },
+    detectSystemFrameRate: function()
+    {
+        let last = 0;
+        let average = 0;
+        let count = 0;
+        const outputField = document.querySelector("#frame-rate-detection span");
+        const finish = function()
+        {
+            const commonFrameRates = [15, 30, 45, 60, 90, 120, 144];
+            const distanceFromFrameRates = commonFrameRates.map(rate => {
+                return Math.abs(Math.round(rate - average));
+            });
+            let targetFrameRate = 60;
+            for (let i = 0; i < commonFrameRates.length; i++) {
+                if (distanceFromFrameRates[i] < 10) {
+                    targetFrameRate = commonFrameRates[i];
+                    break;
+                }
+            }
+            document.getElementById("frame-rate-detection").textContent = `Detected system frame rate as ${targetFrameRate} FPS`;
+            document.getElementById("system-frame-rate").value = targetFrameRate;
+            document.getElementById("frame-rate").value = Math.round(targetFrameRate * 5 / 6);
+        }
+        const tick = function(t)
+        {
+            t = t || 0;
```

Comment:
Can 't' be undefined?

Reply:
No. I removed this. And renamed `t` to `timestamp` to be more clear.